core/arch/x86/decode_table.c

```diff
@@ -1514,6 +1515,7 @@ const instr_info_t * const op_instr[] =
 #define Hh_e TYPE_H, OPSZ_half_16_vex32_evex64
 #define Mes TYPE_M, OPSZ_16_vex32_evex64
 #define Med TYPE_M, OPSZ_16_vex32_evex64
+#define Mex TYPE_M, OPSZ_16_vex32_evex64
```

Comment:
I not sure that we can change `Mex` with `Me`, because Me is fixed 512 byte but we need 128, 256 or 512 bytes.
In user manual operands of vmovntdq marked as `Mx, Vx`. I tried to make similar to it.
Also I do not understand why you defined lot of abbreviations which addressing modes and operand types the same (For example Mes, Med).

Reply:
Good catch, Me is already used. We do need it for TYPE_M, OPSZ_16_vex32_evex64. Please  rename the existing Me into Moq (octo-quad), and redefine Me to above in the AVX-512 section.

> In user manual operands of vmovntdq marked as Mx, Vx. 

These are used for the vex encodings to describe OPSZ_16_vex32. While Intel did logically extend these types to AVX-512, the encodings are different.

> Also I do not understand why you defined lot of abbreviations which addressing modes and operand types the same (For example Mes, Med).

That's because DynamoRIO sticks to the Intel tables but doesn't describe the types in more detail beyond their total size.

Reply:
I have renamed and redefined.

> That's because DynamoRIO sticks to the Intel tables but doesn't describe the types in more detail beyond their total size.
But you have lot of overlapping. Is it normal?

Reply:
Thanks.